src/stream/src/executor/managed_state/aggregation/extreme.rs

```diff
@@ -317,42 +361,26 @@ where
     }
 
     /// Flush the internal state to a write batch.
-    fn flush_inner(&mut self, write_batch: &mut WriteBatch<S>) -> StreamExecutorResult<()> {
-        // Generally, we require the state the be dirty before flushing. However, it is possible
-        // that after a sequence of operations, the flush buffer becomes empty. Then, the
-        // state becomes "dirty", but we do not need to flush anything.
-        if !self.is_dirty() {
-            self.retain_top_n();
-            return Ok(());
-        }
-
-        let mut local = write_batch.prefixify(&self.keyspace);
-        let value_meta = ValueMeta::with_vnode(self.vnode);
-
-        // TODO: we can populate the cache while flushing, but that's hard.
-
-        for ((key, pks), v) in std::mem::take(&mut self.flush_buffer) {
-            let key_encoded = self.serializer.serialize(key, &pks)?;
-            match v.into_option() {
-                Some(v) => {
-                    local.put(
-                        key_encoded,
-                        StorageValue::new_put(
-                            value_meta,
-                            serialize_cell(&v).map_err(StreamExecutorError::serde_error)?,
-                        ),
-                    );
-                }
-                None => {
-                    local.delete_with_value_meta(key_encoded, value_meta);
-                }
-            }
-        }
-
+    fn flush_inner(&mut self) -> StreamExecutorResult<()> {
         self.retain_top_n();
-
         Ok(())
     }
+
+    // TODO: After state table refactored to derive pk from value, should be fixed to only return
+    // value.
+    /// Assemble value for relational table used by extreme state. Should be `group_key` +
+    /// `sort_key` + input pk + agg call value.
+    fn get_relational_value(&self, sort_key: Datum, extreme_pk: ExtremePk) -> Row {
+        let mut sort_key_vec = if let Some(group_key) = self.group_key.as_ref() {
+            group_key.0.to_vec()
+        } else {
+            vec![]
+        };
+        sort_key_vec.push(sort_key.clone());
+        sort_key_vec.extend(extreme_pk.into_iter());
+        sort_key_vec.push(sort_key);
```

Comment:
Do we ensure that this datum is serialized with value-encoding?

Reply:
I've not check the code but this should be done in StateTable.

Reply:
pk serialize is preformed in StateTable and value serialize is performed while writing to state_store.